File: cpp/math/main_10757.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

int main() {
	string a, b;
	cin >> a >> b;

	const int aLen = a.length(), bLen = b.length();
	if (aLen <= 18 && bLen <= 18) {
		cout << stoull(a) + stoull(b) << endl;
		return 0;
	}

	vector<string> sum;
	int aPos = aLen, bPos = bLen;
	int carry = 0;
	while (true) {
		aPos -= 18;
		bPos -= 18;
		unsigned long long aNum = 0, bNum = 0;
		if (aPos >= 0) {
			aNum = stoull(a.substr(aPos, 18));
		} else if (aPos > -18) {
			aNum = stoull(a.substr(0, 18 + aPos));
		}
		if (bPos >= 0) {
			bNum = stoull(b.substr(bPos, 18));
		} else if (bPos > -18) {
			bNum = stoull(b.substr(0, 18 + bPos));
		}
		aNum += (bNum + carry);
		if (aNum >= 1000000000000000000) {
			aNum -= 1000000000000000000;
			carry = 1;
		} else
			carry = 0;
		sum.push_back(to_string((aNum)));
		if (aPos < 0 && bPos < 0) {
			break;
		}
	}
	cout << sum.back();
	sum.pop_back();

	
	for (auto it = sum.rbegin(); it != sum.rend(); ++it) {
		for (int i = it->length(); i < 18; ++i) {
			cout << 0;
		}
		cout << *it;
	}
}

File: kotlin/math/main_10757.kt
fun main() {
    val (a, b) = readLine()!!.split(" ")
    val sum = mutableListOf<String>()
    var carry = 0

    var aPos = a.length - 18
    var bPos = b.length - 18
    while (true) {
        val aNum = when {
                aPos > 0 -> a.substring(aPos, aPos + 18).toULong()
                aPos in -17..0 -> a.substring(0, aPos + 18).toULong()
                else -> 0UL
            }
        val bNum = when {
            bPos > 0 -> b.substring(bPos, bPos + 18).toULong()
            bPos in -17..0 -> b.substring(0, bPos + 18).toULong()
            else -> 0UL
        }

        var s = aNum + bNum + carry.toULong()
        if (s >= 1000000000000000000UL) {
            s -= 1000000000000000000UL
            carry = 1
        } else
            carry = 0

        sum.add(s.toString())
        if (aPos < 0 && bPos < 0)
            break
        aPos -= 18
        bPos -= 18
    }

    print(sum.last())
    for (i in (sum.size - 2).downTo(0))
        print(sum[i].padStart(18, '0'))
}